### PR TITLE
Add timestamp to allocation using PerfClock

### PR DESCRIPTION
**Pull request by nsavoire**
# What does this PR do?

Add timestamp information to allocation events using perf clock.=

Perf clock parameters are transferred to the library through metadata page.

### CODE REVIEW

File: src/ringbuffer_utils.cc [hunk 1]
```diff
@@ -59,6 +75,14 @@ DDRes ring_buffer_create(size_t buffer_size_page_order,
   pevent->custom_event = custom_event;
   pevent->ring_buffer_type = ring_buffer_type;
   pevent->ring_buffer_size = buffer_size;
+
+  if (ring_buffer_type == RingBufferType::kMPSCRingBuffer) {
```
Comment (r1viollet) [inline src/ringbuffer_utils.cc:79]: The workflow is slightly strange as we `mmap` and `munmap` the buffer within the init.
Though as discussed we do not have the mmap within the creation workflow.

Comment (r1viollet) [reply src/ringbuffer_utils.cc:79]: Perhaps we can add a comment to justify the logic

Comment (nsavoire) [reply src/ringbuffer_utils.cc:79]: Done

**Review by r1viollet — APPROVED** [review]
LGTM!
Very cool mechanism to have consistent time information across different event types.

**Pull request merged**